vmupdate/agent/source/pacman/pacman_cli.py

```diff
@@ -0,0 +1,65 @@
+# coding=utf-8
+#
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# Copyright (C) 2022  Piotr Bartman <prbartman@invisiblethingslab.com>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+
+from typing import List
+
+from source.common.package_manager import PackageManager
+from source.common.process_result import ProcessResult
+
+
+class PACMANCLI(PackageManager):
+    def __init__(self, log_handler, log_level):
+        super().__init__(log_handler, log_level)
+        self.package_manager: str = "pacman"
+
+    def refresh(self, hard_fail: bool) -> ProcessResult:
+        """
+        Use package manager to refresh available packages.
+
+        :param hard_fail: raise error if some repo is unavailable
+        :return: (exit_code, stdout, stderr)
+        """
+        cmd = [self.package_manager, "-Sy"]
```

Comment:
This is potentially problematic, as `pacman -Sy` without a related `-Su` or `-Syu` will produce a partial upgrade on the next `pacman -S <pkg>` as it is explained here (second point): https://wiki.archlinux.org/title/System_maintenance#Partial_upgrades_are_unsupported

Now if you can guarantee that there are no installations happening between `refresh()` and the update (`get_action()`?) that is not a problem, but if you cant you might want to consider something like `checkupdates` from `extra/pacman-contrib`, which does the sync against its own set of databases.

Reply:
Yeah I took that into account. 
Initially wanted to make pyalpm sync dbs to a temporary path of it's own to avoid that, but then in my testing there was a very short period between refresh and upgrade where the script releases the lock on the pacman DBs, which I think is highly unlikely to be enough for a package install. 

Either way to be extra safe I can theoretically move the refresh and the upgrade under one transaction, which will hold a lock on pacman DBs. A downside would be that it will make tracking the current progress harder. 
Or I can also just manually create and hold the  db.lock file until the very end of the script.

Reply:
Using checkupdates will also make it harder to track the update progress as we would have to wait for checkupdates to finish without being able to track progress, then do our own sync with progress tracking which would be an unintuitively long period for the end user without any (visible) progress.

Reply:
It somehow completely flew over my head that this is about pacman_cli, which has no progress reporting to begin with, sorry.

I'll experiment with checkupdates and artificially creating an holding a lock on the local DBs to see which works the best here. 

Will update with a patch.

Reply:
Technically speaking, I didn't really see the point of the `refresh()` step when it came to Archlinux in the first place, I just implemented it for the sake of interface completeness.  If @marmarek approves, I think the `refresh()` func for archlinux can be a no-op and it'll both solve this problem and reduce the running time of the whole upgrade.

Reply:
> I think the `refresh()` func for archlinux can be a no-op and it'll both solve this problem and reduce the running time of the whole upgrade.

I'm okay with that.

Reply:
Did just that with the latest commit.